src/main/java/io/druid/data/input/impl/SpatialDimensionSchema.java

```diff
@@ -18,12 +18,14 @@
 package io.druid.data.input.impl;
 
 import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnore;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 import java.util.List;
 
 /**
  */
+@Deprecated
 public class SpatialDimensionSchema
 {
   private final String dimName;
```

Comment:
can't this class be modified to be what NewSpatialDImensionSchema is? You probably want to keep the dimension attribute in json to be "name" instead of "dimName" but it should be possible to support both in this same class by adding a third argument ` @JsonProperty("name") String dimName,` to constructor and initialize dimName from whichever user specified. we can deprecate the "dimName" attribute.